autonomy/cli/scaffold_fsm.py

```diff
@@ -0,0 +1,769 @@
+    return result
+
+
+class AbstractFileGenerator(ABC):
+    """An abstract class for file generators."""
+
+    FILENAME: str
+
+    def __init__(self, ctx: Context, skill_name: str, dfa: DFA) -> None:
+        """Initialize the abstract file generator."""
+        self.ctx = ctx
+        self.skill_name = skill_name
+        self.dfa = dfa
+
+    @abstractmethod
+    def get_file(self) -> str:
+        """Get file content."""
+
+    def write_file(self, output_dir: Path) -> None:
+        """Write the file to output_dir/FILENAME."""
+        (output_dir / self.FILENAME).write_text(self.get_file())
+
+
+class RoundFileGenerator(AbstractFileGenerator):
+    """File generator for 'rounds.py' modules."""
+
+    FILENAME = ROUNDS_FILENAME
+
+    ROUNDS_FILE_HEADER = dedent(
+        """\
+        \"\"\"This package contains the rounds of {FSMName}.\"\"\"
+
+        from enum import Enum
+        from typing import List, Optional, Set, Tuple
+
+        from packages.valory.skills.abstract_round_abci.base import (
+            AbciApp,
+            AbciAppTransitionFunction,
+            AbstractRound,
+            AppState,
+            BaseSynchronizedData,
+            BaseTxPayload,
+            DegenerateRound,
+            EventToTimeout,
+            TransactionType
+        )
+
+    """
+    )
+
+    SYNCHRONIZED_DATA_SECTION = dedent(
+        """\
+        class SynchronizedData(BaseSynchronizedData):
+            \"\"\"
+            Class to represent the synchronized data.
+
+            This data is replicated by the tendermint application.
+            \"\"\"
+
+    """
+    )
+
+    ROUND_CLS_TEMPLATE = dedent(
+        """\
+        class {RoundCls}({ABCRoundCls}):
+            {todo_abstract_round_cls}
+            # TODO: set the following class attributes
+            round_id: str
+            allowed_tx_type: Optional[TransactionType]
+            payload_attribute: str
+
+            def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
+                \"\"\"Process the end of the block.\"\"\"
+                raise NotImplementedError
+
+            def check_payload(self, payload: BaseTxPayload) -> None:
+                \"\"\"Check payload.\"\"\"
+                raise NotImplementedError
+
+            def process_payload(self, payload: BaseTxPayload) -> None:
+                \"\"\"Process payload.\"\"\"
+                raise NotImplementedError
+
+    """
+    )
+
+    ABCI_APP_CLS_TEMPLATE = dedent(
+        """\
+        class {AbciAppCls}(AbciApp[Event]):
+            initial_round_cls: AppState = {initial_round_cls}
+            initial_states: Set[AppState] = {initial_states}
+            transition_function: AbciAppTransitionFunction = {transition_function}
+            final_states: Set[AppState] = {final_states}
+            event_to_timeout: EventToTimeout = {{}}
+            cross_period_persisted_keys: List[str] = []
+    """
+    )
+
+    def get_file(self) -> str:
```

Comment:
something along the lines of `create_file_content` would be more appropriate, as this does not get a file.